kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/KuraIp6Privacy.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm;
+
+import org.eclipse.kura.nm.enums.NMSettingIP6ConfigPrivacy;
+
+public enum KuraIp6Privacy {
+
+    UNKNOWN,
+    DISABLED,
+    ENABLED_PUBLIC_ADD,
+    ENABLED_TEMP_ADD;
+
+    public static KuraIp6Privacy fromString(String status) {
+        switch (status) {
+        case "netIPv6PrivacyUnknown":
+            return KuraIp6Privacy.UNKNOWN;
+        case "netIPv6PrivacyDisabled":
+            return KuraIp6Privacy.DISABLED;
+        case "netIPv6PrivacyEnabledPubAdd":
+            return KuraIp6Privacy.ENABLED_PUBLIC_ADD;
+        case "netIPv6PrivacyEnabledTempAdd":
+            return KuraIp6Privacy.ENABLED_TEMP_ADD;
+        default:
+            throw new IllegalArgumentException(String.format("Unsupported IPv6 privacy value: \"%s\"", status));
+        }
+    }
```

Comment:
Sorry to be back on this but I just noticed:
1. I don't think you need to use have an `unknown` case. Either is disabled or one of the enabled.
2. I would prefer more descriptive strings see below:

```suggestion
    DISABLED,
    ENABLED_PUBLIC_ADD,
    ENABLED_TEMP_ADD;

    public static KuraIp6Privacy fromString(String status) {
        switch (status) {
        case "netIPv6PrivacyDisabled":
            return KuraIp6Privacy.DISABLED;
        case "netIPv6PrivacyEnabledPublicAddress":
            return KuraIp6Privacy.ENABLED_PUBLIC_ADDR;
        case "netIPv6PrivacyEnabledTemporaryAddress":
            return KuraIp6Privacy.ENABLED_TEMP_ADDR;
        default:
            throw new IllegalArgumentException(String.format("Unsupported IPv6 privacy value: \"%s\"", status));
        }
    }
```